spec/unit/pops/types/p_object_type_spec.rb

```diff
@@ -687,6 +675,35 @@ def parse_object(name, body_string)
       expect(eval_and_collect_notices(code)).to eql(['true'])
     end
 
+    it 'a named object is added to the loader irrespective of alias declaration' do
+      code = <<-CODE
+      $x = Object[{ name => 'MyFirstObject' }]
+      notice($x == MyFirstObject)
+      CODE
```

Comment:
I thought about this for a while now.
I think it must be a deliberate act to bind an object type to a type name (or for a type set, if you want to import all of them). I think that is a more important principle than the other things we talked about.
That also means, that it is strange if binding to one name also binds to another (you may not want the object bound to that name).

I think that using a name => in an object time is really only useful when using the type in matches - you want to match on type name. And you would probably never bind those (you cannot actually because you cannot say `type Foo = $x`, which only leaves the example you showed above. There you would simply leave the `Bar` out - why name it something other than what you are binding to? If you want both you can always do:
```
type Foo = Object[{...}]
type Bar = Foo
```